source/custom/nixnet_service.custom.cpp

```diff
@@ -1,2 +1,296 @@
+
+    response->set_status(status);
+    if (status == 0) {
+      void* state_value_raw = (void*)response->state_value().state_value_raw().data();
+      switch (state_id) {
+        case nixnet_grpc::ReadState::READ_STATE_STATE_TIME_CURRENT: {
+          response->mutable_state_value()->set_time_current(*(nxTimestamp100ns_t*)state_value_raw);
+          break;
+        }
+        case nixnet_grpc::ReadState::READ_STATE_STATE_TIME_COMMUNICATING: {
+          response->mutable_state_value()->set_time_communicating(*(nxTimestamp100ns_t*)state_value_raw);
+          break;
+        }
+        case nixnet_grpc::ReadState::READ_STATE_STATE_TIME_START: {
+          response->mutable_state_value()->set_time_start(*(nxTimestamp100ns_t*)state_value_raw);
+          break;
+        }
+        case nixnet_grpc::ReadState::READ_STATE_STATE_CAN_COMM: {
+          SetCanComm(*(u32*)state_value_raw, response->mutable_state_value()->mutable_can_comm());
+          break;
+        }
+        case nixnet_grpc::ReadState::READ_STATE_STATE_FLEX_RAY_COMM: {
+          SetFlexRayComm(*(u32*)state_value_raw, response->mutable_state_value()->mutable_flex_ray_comm());
+          break;
+        }
+        case nixnet_grpc::ReadState::READ_STATE_STATE_LIN_COMM: {
+          SetLinComm((u32*)state_value_raw, response->mutable_state_value()->mutable_lin_comm());
+          break;
+        }
+        case nixnet_grpc::ReadState::READ_STATE_STATE_SESSION_INFO: {
+          response->mutable_state_value()->set_session_info(*(u32*)state_value_raw);
+          break;
+        }
+        case nixnet_grpc::ReadState::READ_STATE_STATE_FLEX_RAY_STATS: {
+          convert_to_grpc(*(_nxFlexRayStats_t*)state_value_raw, response->mutable_state_value()->mutable_flex_ray_stats());
+          break;
+        }
+        case nixnet_grpc::ReadState::READ_STATE_STATE_J1939_COMM: {
+          convert_to_grpc(*(_nxJ1939CommState_t*)state_value_raw, response->mutable_state_value()->mutable_j1939_comm_state());
+          break;
+        }
+        case nixnet_grpc::ReadState::READ_STATE_STATE_TIME_CURRENT_2: {
+          convert_to_grpc(*(_nxTimeLocalNetwork_t*)state_value_raw, response->mutable_state_value()->mutable_time_current2());
+          break;
+        }
+        case nixnet_grpc::ReadState::READ_STATE_STATE_TIME_COMMUNICATING_2: {
+          convert_to_grpc(*(_nxTimeLocalNetwork_t*)state_value_raw, response->mutable_state_value()->mutable_time_communicating2());
+          break;
+        }
+        case nixnet_grpc::ReadState::READ_STATE_STATE_TIME_START_2: {
+          convert_to_grpc(*(_nxTimeLocalNetwork_t*)state_value_raw, response->mutable_state_value()->mutable_time_start2());
+          break;
+        }
+        default: {
+          return ::grpc::Status(::grpc::INVALID_ARGUMENT, "The value for state_id was not specified or out of range");
+          break;
```

Comment:
nit: breaking after a return is dead code.

Reply:
Removed the unnecessary breaks

Reply:
Removed the unnecessary breaks